### PR TITLE
Implement cupy.argpartition

### PR DESCRIPTION
**Pull request by takagi**
Merge #288 first.

This PR implements `cupy.argpartition`.

It uses naive `cupy.argsort` in its implementation, which leads worse performance than ideal one with a proper algorithm but at least satisfies `cupy.argpartition`'s specification.

### CODE REVIEW

**Comment by okuta** [comment]
Cloud you rebase or merge master?

**Comment by takagi** [comment]
jenkins test this please

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -887,7 +888,49 @@ cdef class ndarray:
...
+        if isinstance(kth, int):
+            kth = kth,
+        for k in kth:
+            if k < 0:
+                k += length
+            if not (0 <= k < length):
+                raise ValueError('kth(={}) out of bounds {}'.format(k, length))
+
+        # kth is ignored.
+        return cupy.argsort(data, axis=axis)
```
Comment (unnonouno) [inline cupy/core/core.pyx]: As you says, this implementation is not ideal one. Please write a comment about it.

File: tests/cupy_tests/sorting_tests/test_sort.py [hunk 2]
```diff
@@ -437,3 +437,129 @@ def test_partition_invalid_negative_axis(self, xp):
...
+        kth = 2
+        return self.argpartition(a, kth)
+
+    @testing.for_all_dtypes(no_float16=True, no_bool=True)
+    @testing.numpy_cupy_equal()
+    def test_argpartition_one_dim(self, xp, dtype):
+        a = testing.shaped_random((10,), xp, dtype, 100)
+        kth = 2
+        idx = self.argpartition(a, kth)
+        return idx[kth]
```
Comment (unnonouno) [inline tests/cupy_tests/sorting_tests/test_sort.py]: Can you check its values? I mean `(a[idx[:kth]] < a[idx[kth]]).all()` and `(a[idx[kth]] < a[idx[kth:]]).all()`. Please check it for all test cases.

File: tests/cupy_tests/sorting_tests/test_sort.py [hunk 3]
```diff
@@ -489,3 +489,155 @@ def test_partition_invalid_negative_axis2(self):
...
+    @testing.with_requires('numpy>=1.13')
+    @testing.numpy_cupy_raises()
+    def test_argpartition_invalid_negative_axis1(self, xp):
+        a = testing.shaped_random((2, 2, 2), xp, scale=100)
+        kth = 1
+        axis = -4
+        return self.argpartition(a, kth, axis=axis)
+
+    @testing.with_requires('numpy<1.13')
+    @testing.numpy_cupy_raises(accept=ValueError)
```
Comment (unnonouno) [inline tests/cupy_tests/sorting_tests/test_sort.py]: `accept_error`

**Comment by unnonouno** [comment]
Check a comment.

File: tests/cupy_tests/sorting_tests/test_sort.py (hunk 3 above)
Comment (takagi) [reply tests/cupy_tests/sorting_tests/test_sort.py]: Fixed!

**Comment by takagi** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by takagi** [comment]
Please review again after CI will have passed.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**